apps/openchallenges/edam-etl/src/main.py

```diff
@@ -1,16 +1,68 @@
-"""Extract, transform and load EDAM concepts"""
-
+import pandas as pd
+import requests
 from os import getenv
 
 # Get config from the environment variables
 oc_db_url = getenv("OC_DB_URL")
+version = getenv("OC_DB_VERSION")
+print(f"EDAM Version: {version}")
 print(f"OC DB URL: {oc_db_url}")
 
-# TODO Download EDAM concepts from GitHub or S3 bucket (CSV file)
-print("[TODO] Download EDAM concepts from GitHub or S3 bucket (CSV file)")
 
-# TODO Process the EDAM concepts
-print("[TODO] Process the EDAM concepts")
+# Download EDAM concepts from GitHub or S3 bucket (CSV file)
+def download_edam_csv(url, version):
+    print("Downloading the EDAM concepts from GitHub (CSV file)...")
+    try:
+        response = requests.get(url)
+        response.raise_for_status()  # Raise an exception for bad response
+        with open(f"EDAM_{version}.csv", "wb") as f:
+            f.write(response.content)
+        print("EDAM concepts downloaded successfully.")
+    except requests.RequestException as e:
+        print(f"Error downloading EDAM concepts: {e}")
+
+
+# Transform the CSV to a DataFrame with indices starting from 0
+def transform_to_dataframe(version):
+    print("Processing the EDAM concepts...")
+    try:
+        df = pd.read_csv(f"EDAM_{version}.csv")
+        df["id"] = df.index
+        # Select and rename specific columns
+        df = df[["id", "Class ID", "Preferred Label"]]
+        df = df.rename(
```

Comment:
I'm a big fan of method chaining (which comes with the benefits of streamlining the data processing as well as being more memory-efficient), so we could alternatively do the following instead:

```python
df = (
   pd.read_csv(f"EDAM_{version}.csv", usecols=['Class ID', 'Preferred Label'])
   .rename(columns={"Class ID": "class_id", "Preferred Label": "preferred_label"})
   .assign(id=df.index + 1)
)
```

Here, I added `usecols` to the `read_csv()` method, so that we subset which columns we need rather than load the entire file into the disk space first.  I additionally added `assign()` to the chain, which will add a new column to the data frame called `id` (therefore, replacing line 30).  I also added 1 to `df.index` since I believe "id" needs to start at 1 rather than 0.  @tschaffter to confirm.

We could alternatively do the following as well:

```python
df2 = (
   pd.read_csv("Downloads/EDAM.csv", usecols=["Class ID", "Preferred Label"])
   .reset_index()
   .rename(columns={"index": "id", "Class ID": "class_id", "Preferred Label": "preferred_label"})
   .assign(id=df.index + 1)
)
```

The difference in these two approaches is that `id` would be listed first in `df2`, rather than listed last like in `df`:

```
>>> df.columns
Index(['class_id', 'preferred_label', 'id'], dtype='object')
>>> df2.columns
Index(['id', 'class_id', 'preferred_label'], dtype='object')
```

Reply:
used the first method and made improvements to ensure code executed as it resulted in an erorr related to assigning the id `Error processing EDAM concepts: cannot access local variable 'df' where it is not associated with a value`